packages/dev/core/src/Materials/Node/nodeMaterialBuildState.ts

```diff
@@ -192,7 +211,11 @@ export class NodeMaterialBuildState {
      */
     public _emit2DSampler(name: string) {
         if (this.samplers.indexOf(name) < 0) {
-            this._samplerDeclaration += `uniform sampler2D ${name};\n`;
+            if (this.shaderLanguage === ShaderLanguage.WGSL) {
+                this._samplerDeclaration += `var ${name}: sampler;\n`;

```

Comment:
You will also have to generate a texture object, so something like:
```glsl
this._samplerDeclaration += `var ${name}Sampler: sampler;\n`;
this._samplerDeclaration += `var ${name}Texture: texture_2d<f32>;\n`;
```
It assumes the type is `f32`, but I don't think we can use other types in the NME (yet?).